src/contracts/collections/pool_keys.rs

```diff
@@ -1,41 +1,65 @@
 use crate::{contracts::PoolKey, InvariantError};
 use alloc::vec::Vec;
-
+use ink::storage::Mapping;
 #[ink::storage_item]
 #[derive(Debug, Default)]
 pub struct PoolKeys {
-    pool_keys: Vec<PoolKey>,
+    pool_keys: Mapping<PoolKey, u16>,
+    pool_keys_by_index: Mapping<u16, PoolKey>,
+    pool_keys_length: u16,
 }
 
 impl PoolKeys {
+    pub fn get(&self, pool_key: PoolKey) -> Option<u16> {
+        self.pool_keys.get(pool_key)
+    }
+
     pub fn add(&mut self, pool_key: PoolKey) -> Result<(), InvariantError> {
         if self.contains(pool_key) {
             return Err(InvariantError::PoolKeyAlreadyExist);
         }
 
-        self.pool_keys.push(pool_key);
+        self.pool_keys.insert(pool_key, &self.pool_keys_length);
+        self.pool_keys_by_index
+            .insert(self.pool_keys_length, &pool_key);
+        self.pool_keys_length += 1;
+
         Ok(())
     }
 
     #[allow(dead_code)]
     pub fn remove(&mut self, pool_key: PoolKey) -> Result<(), InvariantError> {
-        let index = self
-            .pool_keys
-            .iter()
-            .position(|vec_pool_key| *vec_pool_key == pool_key)
-            .ok_or(InvariantError::PoolKeyNotFound)?;
+        let extracted_pool_key = self.get(pool_key);
+        if extracted_pool_key.is_none() {
+            return Err(InvariantError::PoolKeyNotFound);
+        }
 
-        self.pool_keys.remove(index);
+        let pool_key_index = extracted_pool_key.unwrap();
 
+        self.pool_keys.remove(pool_key);
+        self.pool_keys_by_index.remove(pool_key_index);
+        self.pool_keys_length -= 1;
         Ok(())
     }
 
-    pub fn contains(&self, fee_tier_key: PoolKey) -> bool {
-        self.pool_keys.contains(&fee_tier_key)
+    pub fn contains(&self, pool_key: PoolKey) -> bool {
+        self.pool_keys.get(pool_key).is_some()
     }
 
-    pub fn get_all(&self) -> Vec<PoolKey> {
-        self.pool_keys.clone()
+    pub fn get_all(&self, size: u8) -> Result<Vec<PoolKey>, InvariantError> {
+        if size > 220 {
```

Comment:
The best option is to save this value (`220`) as a constant with an explanation of the calculations, similar to the size limits we previously defined